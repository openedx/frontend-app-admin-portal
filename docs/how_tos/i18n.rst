####################
React App i18n HOWTO
####################

This document has moved to the frontend-platform repo: https://github.com/openedx/frontend-platform/blob/master/docs/how_tos/i18n.rst
