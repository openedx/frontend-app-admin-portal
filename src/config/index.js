import { hasFeatureFlagEnabled } from '@edx/frontend-enterprise-utils';

const configuration = {
  BASE_URL: process.env.BASE_URL,
  LMS_BASE_URL: process.env.LMS_BASE_URL,
  LOGIN_URL: process.env.LOGIN_URL,
  LOGOUT_URL: process.env.LOGOUT_URL,
  ENTERPRISE_SUPPORT_URL: process.env.ENTERPRISE_SUPPORT_URL,
  ENTERPRISE_SUPPORT_REVOKE_LICENSE_URL: process.env.ENTERPRISE_SUPPORT_REVOKE_LICENSE_URL,
  SURVEY_MONKEY_URL: process.env.SURVEY_MONKEY_URL,
  CSRF_TOKEN_API_PATH: process.env.CSRF_TOKEN_API_PATH,
  REFRESH_ACCESS_TOKEN_ENDPOINT: process.env.REFRESH_ACCESS_TOKEN_ENDPOINT,
  DATA_API_BASE_URL: process.env.DATA_API_BASE_URL,
  ECOMMERCE_BASE_URL: process.env.ECOMMERCE_BASE_URL,
  LICENSE_MANAGER_BASE_URL: process.env.LICENSE_MANAGER_BASE_URL,
  DISCOVERY_BASE_URL: process.env.DISCOVERY_BASE_URL,
  SECURE_COOKIES: process.env.NODE_ENV !== 'development',
  SEGMENT_KEY: process.env.SEGMENT_KEY,
  ACCESS_TOKEN_COOKIE_NAME: process.env.ACCESS_TOKEN_COOKIE_NAME,
  USER_INFO_COOKIE_NAME: process.env.USER_INFO_COOKIE_NAME,
  NODE_ENV: process.env.NODE_ENV,
  CUSTOMER_SUPPORT_EMAIL: 'customersuccess@edx.org', // TODO: avoid using hardcoded email address here...
  TABLEAU_URL: process.env.TABLEAU_URL,
  ENTERPRISE_LEARNER_PORTAL_URL: process.env.ENTERPRISE_LEARNER_PORTAL_URL,
  ALGOLIA: {
    APP_ID: process.env.ALGOLIA_APP_ID,
    SEARCH_API_KEY: process.env.ALGOLIA_SEARCH_API_KEY,
    INDEX_NAME: process.env.ALGOLIA_INDEX_NAME,
  },
  LOGO_URL: process.env.LOGO_URL,
  LOGO_WHITE_URL: process.env.LOGO_WHITE_URL,
  LOGO_TRADEMARK_URL: process.env.LOGO_TRADEMARK_URL,
};

const features = {
  BULK_ENROLLMENT: process.env.FEATURE_BULK_ENROLLMENT || hasFeatureFlagEnabled('BULK_ENROLLMENT'),
  CODE_MANAGEMENT: process.env.FEATURE_CODE_MANAGEMENT || hasFeatureFlagEnabled('CODE_MANAGEMENT'),
  REPORTING_CONFIGURATIONS: process.env.FEATURE_REPORTING_CONFIGURATIONS || hasFeatureFlagEnabled('REPORTING_CONFIGURATIONS'),
  ANALYTICS: process.env.FEATURE_ANALYTICS || hasFeatureFlagEnabled('ANALYTICS'),
  SAML_CONFIGURATION: process.env.FEATURE_SAML_CONFIGURATION || hasFeatureFlagEnabled('SAML_CONFIGURATION'),
  SUPPORT: process.env.FEATURE_SUPPORT || hasFeatureFlagEnabled('SUPPORT'),
  EXTERNAL_LMS_CONFIGURATION: process.env.FEATURE_EXTERNAL_LMS_CONFIGURATION || hasFeatureFlagEnabled('EXTERNAL_LMS_CONFIGURATION'),
  FILE_ATTACHMENT: process.env.FEATURE_FILE_ATTACHMENT || hasFeatureFlagEnabled('FILE_ATTACHMENT'),
  SETTINGS_PAGE: process.env.FEATURE_SETTINGS_PAGE || hasFeatureFlagEnabled('SETTINGS_PAGE'),
  SETTINGS_UNIVERSAL_LINK: process.env.FEATURE_SETTINGS_UNIVERSAL_LINK || hasFeatureFlagEnabled('SETTINGS_UNIVERSAL_LINK'),
  FEATURE_BROWSE_AND_REQUEST: process.env.FEATURE_BROWSE_AND_REQUEST || hasFeatureFlagEnabled('BROWSE_AND_REQUEST'),
};

export { configuration, features };
