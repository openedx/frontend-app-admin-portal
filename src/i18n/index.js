import { messages as paragonMessages } from '@edx/paragon';

export default [
  paragonMessages,
];
