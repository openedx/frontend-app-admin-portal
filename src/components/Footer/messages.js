import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  termsOfService: {
    id: 'adminPortal.emailTemplateForm.termsOfService',
    defaultMessage: 'Terms of Service',
  },
  privacyPolicy: {
    id: 'adminPortal.emailTemplateForm.privacyPolicy',
    defaultMessage: 'Privacy Policy',
  },
  support: {
    id: 'adminPortal.emailTemplateForm.support',
    defaultMessage: 'Support',
  },
});

export default messages;
