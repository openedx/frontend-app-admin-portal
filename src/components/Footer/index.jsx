import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';

import { configuration } from '../../config';

import Img from '../Img';
import messages from './messages';
import './Footer.scss';

class Footer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      enterpriseLogoNotFound: false,
    };
  }

  componentDidUpdate(prevProps) {
    const { enterpriseLogo } = this.props;
    if (enterpriseLogo && enterpriseLogo !== prevProps.enterpriseLogo) {
      this.setState({ // eslint-disable-line react/no-did-update-set-state
        enterpriseLogoNotFound: false,
      });
    }
  }

  renderEnterpriseLogo() {
    const { enterpriseLogo, enterpriseSlug, enterpriseName } = this.props;
    return (
      <Link className="logo pl-4" to={`/${enterpriseSlug}`}>
        <Img
          src={enterpriseLogo}
          alt={`${enterpriseName} logo`}
          onError={() => this.setState({ enterpriseLogoNotFound: true })}
        />
      </Link>
    );
  }

  render() {
    const { enterpriseLogoNotFound } = this.state;
    const { enterpriseLogo } = this.props;
    const { formatMessage } = this.props.intl;

    return (
      <footer className="container-fluid py-4 border-top">
        <div className="row justify-content-between align-items-center">
          <div className="col-xs-12 col-md-4 logo-links">
            <Link className="logo border-right pr-4" to="/">
              <Img src={configuration.LOGO_TRADEMARK_URL} alt="edX logo" />
            </Link>
            {enterpriseLogo && !enterpriseLogoNotFound && this.renderEnterpriseLogo()}
          </div>
          <div className="col-xs-12 col-md footer-links">
            <nav>
              <ul className="nav justify-content-end small">
                <li className="nav-item border-right">
                  <a className="nav-link px-2" href="https://www.edx.org/edx-terms-service">
                    {formatMessage(messages.termsOfService)}
                  </a>
                </li>
                <li className="nav-item border-right">
                  <a className="nav-link px-2" href="https://www.edx.org/edx-privacy-policy">
                    {formatMessage(messages.privacyPolicy)}
                  </a>
                </li>
                <li className="nav-item">
                  <Link
                    className="nav-link px-2"
                    to={configuration.ENTERPRISE_SUPPORT_URL}
                    target="_blank"
                  >
                    {formatMessage(messages.support)}
                  </Link>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </footer>
    );
  }
}

Footer.propTypes = {
  enterpriseName: PropTypes.string,
  enterpriseSlug: PropTypes.string,
  enterpriseLogo: PropTypes.string,
  intl: intlShape.isRequired, // injected by injectIntl
};

Footer.defaultProps = {
  enterpriseName: null,
  enterpriseSlug: null,
  enterpriseLogo: null,
};

export default injectIntl(Footer);
