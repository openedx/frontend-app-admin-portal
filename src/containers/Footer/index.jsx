import { connect } from 'react-redux';

import Footer from '../../components/Footer';

const mapStateToProps = state => ({
  enterpriseName: state.portalConfiguration.enterpriseName,
  enterpriseSlug: state.portalConfiguration.enterpriseSlug,
  enterpriseLogo: state.portalConfiguration.enterpriseBranding?.logo,
});

export default connect(mapStateToProps)(Footer);
