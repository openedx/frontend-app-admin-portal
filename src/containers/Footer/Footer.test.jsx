import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { IntlProvider } from '@edx/frontend-platform/i18n';
import { configuration } from '../../config';

import Footer from './index';

const mockStore = configureMockStore([thunk]);

const FooterWrapper = props => (
  <MemoryRouter>
    <IntlProvider locale="en">
      <Footer
        store={props.store}
        {...props}
      />
    </IntlProvider>
  </MemoryRouter>
);

FooterWrapper.propTypes = {
  store: PropTypes.shape({}).isRequired,
};

describe('<Footer />', () => {
  let store;
  let tree;

  it('renders enterprise logo correctly', () => {
    store = mockStore({
      portalConfiguration: {
        enterpriseName: 'Test Enterprise',
        enterpriseSlug: 'test-enterprise',
        enterpriseBranding: {
          logo: 'https://test.url/image/1.png',
        },
      },
    });

    tree = renderer
      .create((
        <FooterWrapper store={store} />
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders edX logo correctly', () => {
    store = mockStore({
      portalConfiguration: {},
    });
    tree = renderer
      .create((
        <FooterWrapper store={store} />
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correct help center link from config', () => {
    configuration.ENTERPRISE_SUPPORT_URL = 'http://test-hc.com/hc';
    store = mockStore({
      portalConfiguration: {},
    });
    tree = renderer
      .create((
        <FooterWrapper store={store} />
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
