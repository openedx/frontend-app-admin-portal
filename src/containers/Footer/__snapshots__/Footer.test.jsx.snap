// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correct help center link from config 1`] = `
<footer
  className="container-fluid py-4 border-top"
>
  <div
    className="row justify-content-between align-items-center"
  >
    <div
      className="col-xs-12 col-md-4 logo-links"
    >
      <a
        className="logo border-right pr-4"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="edX logo"
          src="https://edx-cdn.org/v3/default/logo-trademark.svg"
        />
      </a>
    </div>
    <div
      className="col-xs-12 col-md footer-links"
    >
      <nav>
        <ul
          className="nav justify-content-end small"
        >
          <li
            className="nav-item border-right"
          >
            <a
              className="nav-link px-2"
              href="https://www.edx.org/edx-terms-service"
            >
              Terms of Service
            </a>
          </li>
          <li
            className="nav-item border-right"
          >
            <a
              className="nav-link px-2"
              href="https://www.edx.org/edx-privacy-policy"
            >
              Privacy Policy
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link px-2"
              href="http://test-hc.com/hc"
              target="_blank"
            >
              Support
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</footer>
`;

exports[`<Footer /> renders edX logo correctly 1`] = `
<footer
  className="container-fluid py-4 border-top"
>
  <div
    className="row justify-content-between align-items-center"
  >
    <div
      className="col-xs-12 col-md-4 logo-links"
    >
      <a
        className="logo border-right pr-4"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="edX logo"
          src="https://edx-cdn.org/v3/default/logo-trademark.svg"
        />
      </a>
    </div>
    <div
      className="col-xs-12 col-md footer-links"
    >
      <nav>
        <ul
          className="nav justify-content-end small"
        >
          <li
            className="nav-item border-right"
          >
            <a
              className="nav-link px-2"
              href="https://www.edx.org/edx-terms-service"
            >
              Terms of Service
            </a>
          </li>
          <li
            className="nav-item border-right"
          >
            <a
              className="nav-link px-2"
              href="https://www.edx.org/edx-privacy-policy"
            >
              Privacy Policy
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link px-2"
              href="/"
              onClick={[Function]}
              target="_blank"
            >
              Support
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</footer>
`;

exports[`<Footer /> renders enterprise logo correctly 1`] = `
<footer
  className="container-fluid py-4 border-top"
>
  <div
    className="row justify-content-between align-items-center"
  >
    <div
      className="col-xs-12 col-md-4 logo-links"
    >
      <a
        className="logo border-right pr-4"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="edX logo"
          src="https://edx-cdn.org/v3/default/logo-trademark.svg"
        />
      </a>
      <a
        className="logo pl-4"
        href="/test-enterprise"
        onClick={[Function]}
      >
        <img
          alt="Test Enterprise logo"
          onError={[Function]}
          src="https://test.url/image/1.png"
        />
      </a>
    </div>
    <div
      className="col-xs-12 col-md footer-links"
    >
      <nav>
        <ul
          className="nav justify-content-end small"
        >
          <li
            className="nav-item border-right"
          >
            <a
              className="nav-link px-2"
              href="https://www.edx.org/edx-terms-service"
            >
              Terms of Service
            </a>
          </li>
          <li
            className="nav-item border-right"
          >
            <a
              className="nav-link px-2"
              href="https://www.edx.org/edx-privacy-policy"
            >
              Privacy Policy
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link px-2"
              href="/"
              onClick={[Function]}
              target="_blank"
            >
              Support
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</footer>
`;
